wagtail/admin/filters.py

```diff
@@ -87,7 +87,38 @@ class FilteredModelChoiceFilter(django_filters.ModelChoiceFilter):
     field_class = FilteredModelChoiceField
 
 
+class LocaleFilter(django_filters.ChoiceFilter):
+    def filter(self, qs, language_code):
+        if language_code:
+            return qs.filter(locale__language_code=language_code)
+        return qs
+
+
 class WagtailFilterSet(django_filters.FilterSet):
+    @classmethod
+    def get_filters(cls):
+        from wagtail.models.i18n import TranslatableMixin
+
+        filters = super().get_filters()
+
+        # Add a locale filter if the model is translatable
+        # and there isn't one already.
+        if (
```

Comment:
question: should this not be behind `getattr(settings, "WAGTAIL_I18N_ENABLED", False)` ?
that is, we don't want the locale filter if this is a single-language setup

Reply:
Good point, although I now have one test failing because the method is called at import time (django-filter's filterset works this way), and `override_settings` has no effect because it's called afterwards :(

Reply:
OK, done in e27e9e594e4e09ca61447f04b0498337fdade944. Had to move the override to `__init__`.

Reply:
I've now extracted it into a separate method in 288b47d. Should make it easier to override as well.